File: code/sdk/ios/OpenCloudSheet/Sources/OpenCloudSheet/Protocols/IPlatformTypeInfo.swift
//
//  IPlatformTypeInfo.swift
//  OpenCloudSheet
//
//  Protocol for providing platform-specific type information to enable cross-platform workbook compatibility
//

import Foundation

public protocol IPlatformTypeInfo {
    static func getIosClassName() -> String
    static func getAndroidClassName() -> String
}

public extension IPlatformTypeInfo {
    static func getIosClassName() -> String {
        return String(reflecting: Self.self)
    }
}

File: code/sdk/android/cloudsheet-Android-sdk/src/main/kotlin/com/opencloudsheet/protocols/IPlatformTypeInfo.kt
package com.opencloudsheet.protocols

/**
 * Interface for providing platform-specific type information to enable cross-platform workbook compatibility.
 *
 * Implementing classes should provide both iOS and Android class names to ensure workbooks
 * can be accessed from both platforms.
 */
interface IPlatformTypeInfo {
    fun getIosClassName(): String
    fun getAndroidClassName(): String = this::class.java.name
}
